### ISSUE TITLE
Why is the app gone from app store?

### ISSUE THREAD

**Issue by lastmetroid** [dc297/mqttclpro#91]
where to get it?

**Comment by petermg** [comment]
Yeah man what happened!?!?

**Comment by blacklight** [comment]
I'm not sure what's happening here, but I've seen this app disappear and reappear on the Play Store quite a few times.

And since it hasn't been updated in 5 years it's probably safe to assume that it's not longer maintained.

Since I rely on this app for quite a lot of tasks on my Tasker installations, I've decided that I can't afford to lose it and [I've made a fork](https://github.com/blacklight/mqttclpro) where I'm trying to get the source code back to work.

Since the codebase hasn't been touched in 5 years there are quite a lot of broken things - namely the minimum Android API is no longer compatible with the minimum supported API for the Play Store (and it may be the reason why it disappeared again recently), the Google Ads SDK used is broken and unmaintained (and I've diligently proceeded with removing all ads-related things on my fork), and, most importantly, the ORM framework used by this app ([Requery](https://github.com/requery/requery)) is now [unmaintained](https://github.com/requery/requery/issues/926) and its annotations API is [incompatible](https://github.com/requery/requery/issues/926#issuecomment-1383840137) with recent versions of Java, so all the `MessageEntity`/`TopicEntity` etc. classes are not even generated anymore.

Long story short, follow my fork or contribute to it if you're interested in trying to get this app back to build on recent versions of Android. But be aware that at the current stage it doesn't build yet. I'll probably keep trying working on it when I have time, but I can't make promises. The biggest hurdle in the way is the Requery dependency. If that's unmaintained and broken, then this app probably needs a major rewrite by changing ORM framework, data model and everything that interacts with the db.

So by now your best bet is, if you still have the app installed on a rooted device (luckily I do), to backup the APK from `/data/app/in.dc297.mqttclpro-<app-id>/base.apk` and keep using that.

**Comment by stainlessray** [comment]
> I'm not sure what's happening here, but I've seen this app disappear and reappear on the Play Store quite a few times.
> 
> And since it hasn't been updated in 5 years it's probably safe to assume that it's not longer maintained.
> 
> Since I rely on this app for quite a lot of tasks on my Tasker installations, I've decided that I can't afford to lose it and [I've made a fork](https://github.com/blacklight/mqttclpro) where I'm trying to get the source code back to work.
> 
> Since the codebase hasn't been touched in 5 years there are quite a lot of broken things - namely the minimum Android API is no longer compatible with the minimum supported API for the Play Store (and it may be the reason why it disappeared again recently), the Google Ads SDK used is broken and unmaintained (and I've diligently proceeded with removing all ads-related things on my fork), and, most importantly, the ORM framework used by this app ([Requery](https://github.com/requery/requery)) is now [unmaintained](https://github.com/requery/requery/issues/926) and its annotations API is [incompatible](https://github.com/requery/requery/issues/926#issuecomment-1383840137) with recent versions of Java, so all the `MessageEntity`/`TopicEntity` etc. classes are not even generated anymore.
> 
> Long story short, follow my fork or contribute to it if you're interested in trying to get this app back to build on recent versions of Android. But be aware that at the current stage it doesn't build yet. I'll probably keep trying working on it when I have time, but I can't make promises. The biggest hurdle in the way is the Requery dependency. If that's unmaintained and broken, then this app probably needs a major rewrite by changing ORM framework, data model and everything that interacts with the db.
> 
> So by now your best bet is, if you still have the app installed on a rooted device (luckily I do), to backup the APK from `/data/app/in.dc297.mqttclpro-<app-id>/base.apk` and keep using that.

Do you plan to continue to try & get it working? I could probably be talked into contributing. Or possibly bootstrapping a replacement? Not looking for a "high engagement" side project, but def always open to helping get a real MQTT tool on Android properly.

**Comment by Zhuinden** [comment]
@blacklight did you succeed?

**Comment by blacklight** [comment]
@Zhuinden I haven't had much time to dedicate to this endeavor lately. My fork already has the changes that remove the ads and the whole outdated Google Play Store dependencies, but there's definitely a lot of work involved in updating the storage model, so if someone can help on that I'm definitely happy to resume the work.

In the meantime I have migrated all of my Tasker scripts that were using this app to use my [ntfy](https://github.com/binwiederhier/ntfy) server instead (publishing works over a standard HTTP POST request and subscribing involves [listening on an intent](`https://docs.ntfy.sh/subscribe/phone/#automation-apps), no other apps required), so I don't really need this app anymore. I can still help on it as a side-side-side project though :)

If you really need MQTT then [Join](https://tasker.joaoapps.com/join.html) by João Dias is a good alternative, but it's not open-source and it's not an option if you have a more sophisticated MQTT setup (like using TLS or authentication).
